client/src/components/CoinInput.tsx

```diff
@@ -73,9 +87,20 @@ export function useCoinInput({
     if (initialAmount != null) setAmount(initialAmount);
   }, [initialAmount]);
 
+  // TODO: consider real gas fee, replacing GAS_FEE variable.
+  // For now we need to keep 1 unit in the wallet(it's not spent) in order to complete "create pool" transaction.
+  function getSafeMaxBalance() {
+    const amount = coinBalance?.amount || BigInt(0);
```

Comment:
You can get amount directly here;

```suggestion
    const amount = amount || BigInt(0);
```

Reply:
No, they're different. `coinBalance?.amount` has balance from wallet, `amount` has input value